material/plugins/info/patterns.py

```diff
@@ -0,0 +1,27 @@
+def get_exclusion_patterns():
+    """
+    Regex patterns, which will be compared against directory and file names
+    case-sensitively. Used with the https://pypi.org/project/regex/ module.
+
+    Additional remarks for pattern creation:
+    - The compared paths will be always in POSIX format.
+    - Each directory path will have a / at the end to allow to distinguish them
+      from files.
+    - Patterns for dynamic or custom paths like Virtual Environments (venv) or
+      build site directories are created during plugin runtime.
+    """
+    return [
+        r".*__pycache__/",  # Python cache directory
+        #
+        r".*\.DS_Store",    # macOS
+        #
+        r"^/\.[^/]+$",      # .dotfiles in the root directory
```

Comment:
This is risky. I can think of one case where this might not work: the awesome pages plugin with the `docs_dir` set to the root. I think we should remove this line or better specify what to exclude.

Reply:
AFAIK MkDocs doesn't allow to set `docs_dir: .`. Do you know if there are any plans to change it?
Also `awesome-pages` looks only within the `docs_dir`, so root of the project should be fine.
```bash
ERROR   -  Config value 'docs_dir': The 'docs_dir' should not be the parent directory of the config file.
           Use a child directory instead so that the 'docs_dir' is a sibling of the config file.
```
I also don't think there are any other cases where a `.dotfile` is used with a plugin. Any files related to theme's plugins are in the `docs_dir` too, like `.meta.yml` or `.authors.yml`. The social layouts are not a `.dotfile`. So the user would have to deliberately break the "recommended way to to things".

I'll see how feasible the explicit patterns for the file would be 🤔 
This repository has 9 .dotfiles, but some of them are related to TypeScript, so perhaps not everything is required, or there are even more "standard" files people use 😵 💫

Reply:
Perhaps it's time to add `inclusion_patterns` as well 😆

Reply:
I kept the exclusion only logic, modified the match function to search to match subdirectories without having to resort to `.*` at the beginning. I still have to properly test it though 😅, a multilayered project with nested .git repositories. This also kind of forced me to use exact names, because `[^/]*lint[^/]*` would match a documentation file `/To lint or not to lint.md`, and adding exclusions for `.md` didn't seem right...